volatility3/framework/plugins/windows/crashinfo.py

```diff
@@ -0,0 +1,83 @@
+# This file is Copyright 2021 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+import datetime
+from volatility3.framework import interfaces, renderers
+from volatility3.framework.configuration import requirements
+from volatility3.framework.renderers import format_hints, conversion
+from volatility3.framework.objects import utility
+
+vollog = logging.getLogger(__name__)
+
+class Crashinfo(interfaces.plugins.PluginInterface):
+    _required_framework_version = (1, 0, 0)
```

Comment:
Technically we should be bumping the middle version triplet on core and this plugin should be depending on that version.  Don't know if you want to go that far, but since it needs the changes to the crashdump layer that you made, this strictly won't work with earlier versions.  I'm keen that we get comfortable and familiar doing that (and therefore making our changes additions rather than changes), but I'm happy to do it gently and therefore let this one slide whilst people are getting used to the idea...

Reply:
Let's leave it - this is on a branch so there shouldn't be any plugins depending on it yet.

Reply:
Sorry, I'm not sure I was clear.  Nothing depends on `crashinfo` yet, because it's not yet been released, however, `crashinfo` itself depends on the changes we're making to the crash layer.  As such, the `constants.py` MINOR version should be bumped (checking that it's only an additive change, and doesn't affect the framework for anyone that might've written something else) and then the _required framework version_ for this plugin would need to say it needs at least that version to work.  Sorry, there's a lot of version numbers flying around, but essentially, there's the following chain:

```
Core: includes layers and anything else that's unversioned
Plugins: and other versioned components
Anything dependent on a versioned component
```
It all ends up making a big dependency graph so that we have some granularity over stuff, but sadly core currently still contains layers and other bits of the core API (the main interfaces).  We could consider versioning the layers, but then each plugin will have to say which versions of which layers it expects.

Hopefully that makes more sense now?  I've unresolved this, but if you'd prefer to sneak this in and handler potential breakages we can do that.  It will be important to get used to the version numbers and how to handle them down the road though, and since this is an additive change, it should be pretty painless...

Reply:
This should be resolved in f1079e7 if I understood correctly. I think the `constants.py` reference was meant to be `constants/__init__.py` but let me know if that is not right.

Reply:
Yep, you're quite right, and the change looks good, thanks!  5:)